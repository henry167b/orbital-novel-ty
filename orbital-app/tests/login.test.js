import React from "react";
import renderer from "react-test-renderer";

import Login from "../app/(auth)/login";

const mockSignIn = jest.fn();

jest.mock('../contexts/auth', () => ({
  useAuth: () => ({
    signIn: mockSignIn,
  }),
}));

jest.mock('../async_storage/storage', () => ({
  clearRecentSearches: jest.fn(),
}));

// jest.mock('react-native-webview', () => 'WebView');



describe("<Login />", () => {
  it("can render", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 1 child", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree[0].children.length).toBe(2);
  })
});

// describe('Login', () => {
//   it('bypass works', () => {
//     const component = renderer.create(<Login />);
//     const instance = component.root;

//     // Get the username and password TextInput components
//     const usernameInput = instance.findByProps({ testID: 'username-input' });
//     const passwordInput = instance.findByProps({ testID: 'password-input' });

//     // Set invalid login details
//     usernameInput.props.onChangeText('novelty');
//     passwordInput.props.onChangeText('password');

//     // Get the SIGN IN button component
//     const signInButton = instance.findByProps({ testID: 'button' });

//     // Trigger onPress event of the SIGN IN button
//     signInButton.props.onPress();

//     // Assert that the error message is displayed
//     expect(mockSignIn).toHaveBeenCalledWith('novelty', 'password');
//   });

//   it('should bypass login for valid preset user', () => {
//     const component = renderer.create(<Login />);
//     const instance = component.root;

//     // Get the username and password TextInput components
//     const usernameInput = instance.findByProps({ testID: 'username-input' });
//     const passwordInput = instance.findByProps({ testID: 'password-input' });

//     // Set valid preset login details
//     usernameInput.props.onChangeText('novelty');
//     passwordInput.props.onChangeText('password');

//     // Get the SIGN IN button component
//     const signInButton = instance.findByProps({ testID: 'button' });

//     // Trigger onPress event of the SIGN IN button
//     signInButton.props.onPress();

//     // Get the error message component
//     const errorMessage = instance.findByProps({ testID: 'error-message' });

//     // Assert that the error message is not displayed
//     expect(errorMessage).toBeUndefined();

//     // Get the loading indicator component
//     const loadingIndicator = instance.findByProps({ testID: 'loading-indicator' });

//     // Assert that the loading indicator is not displayed
//     expect(loadingIndicator).toBeUndefined();

//     // Get the WebView component
//     const webView = instance.findByProps({ testID: 'webview' });

//     // Assert that the WebView is present (bypassed login)
//     expect(webView).toBeTruthy();
//   });
// });