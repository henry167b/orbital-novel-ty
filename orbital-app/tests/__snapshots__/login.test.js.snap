// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> can render 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 10,
        "justifyContent": "space-between",
        "paddingBottom": 25,
        "paddingHorizontal": 40,
        "paddingTop": 100,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
          "gap": 50,
          "justifyContent": "space-evenly",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "System",
              "fontSize": 24,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 32,
            },
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
      >
        Let's Sign You In
      </Text>
      <View
        style={
          {
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              {
                "alignItems": "flex-start",
                "color": "gray",
              },
            ]
          }
        >
          myLibrary User
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "rgba(231, 224, 236, 1)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              {
                "backgroundColor": "white",
                "width": "100%",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(28, 27, 31, 1)",
                "bottom": 0,
                "height": 1,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "transform": [
                  {
                    "scaleY": 0.5,
                  },
                ],
                "zIndex": 1,
              }
            }
            testID="text-input-underline"
          />
          <View
            style={
              [
                {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                {
                  "minHeight": 40,
                },
              ]
            }
          >
            <TextInput
              editable={true}
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="rgba(73, 69, 79, 1)"
              selectionColor="rgba(103, 80, 164, 1)"
              style={
                [
                  {
                    "margin": 0,
                  },
                  {
                    "paddingLeft": 16,
                    "paddingRight": 16,
                  },
                  {
                    "height": 40,
                  },
                  {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  {
                    "color": "rgba(73, 69, 79, 1)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  [
                    {},
                  ],
                  undefined,
                ]
              }
              testID="username-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              {
                "alignItems": "flex-start",
                "color": "gray",
              },
            ]
          }
        >
          Password
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "rgba(231, 224, 236, 1)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              {
                "backgroundColor": "white",
                "width": "100%",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(28, 27, 31, 1)",
                "bottom": 0,
                "height": 1,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "transform": [
                  {
                    "scaleY": 0.5,
                  },
                ],
                "zIndex": 1,
              }
            }
            testID="text-input-underline"
          />
          <View
            style={
              [
                {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                {
                  "minHeight": 40,
                },
              ]
            }
          >
            <TextInput
              editable={true}
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="rgba(73, 69, 79, 1)"
              secureTextEntry={true}
              selectionColor="rgba(103, 80, 164, 1)"
              style={
                [
                  {
                    "margin": 0,
                  },
                  {
                    "paddingLeft": 16,
                    "paddingRight": 16,
                  },
                  {
                    "height": 40,
                  },
                  {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  {
                    "color": "rgba(73, 69, 79, 1)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  [
                    {},
                  ],
                  undefined,
                ]
              }
              testID="password-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(103, 80, 164, 1)",
          "borderRadius": 5,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": "100%",
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "borderColor": "transparent",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 5,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {},
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    {
                      "marginHorizontal": 24,
                      "marginVertical": 10,
                    },
                    undefined,
                    false,
                    {
                      "color": "rgba(255, 255, 255, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
              testID="button-text"
            >
              SIGN IN
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      [
        {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      incognito={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={true}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        {
          "uri": "https://www.nlb.gov.sg/mylibrary",
        }
      }
      style={
        [
          {
            "flex": 1,
            "overflow": "hidden",
          },
          {
            "backgroundColor": "#ffffff",
          },
          {
            "flex": 0,
          },
        ]
      }
      textInteractionEnabled={true}
      useSharedProcessPool={true}
    />
  </View>,
]
`;
